### ISSUE TITLE
Increase Tank Missile Armor?

### ISSUE THREAD

**Issue by KrewGR** [TheSuperHackers/GeneralsGamePatch#1797]
I have been doing some testing and wanted to present my Theory as well as some numbers about the role of tanks in zerohour.

Currently, there isn't a single unit that fills the role of damage soaking. Naturally, Tanks would fill this role in every single RTS, however in zerohour, they are completely nullified due to how useless they are at their role. Even with Tank general, players found the meta of Gats/Dragons/ Lix spam vs Armies such as stealth, which has NO tanks at all. This just doesn't make sense.

The main problem tanks are having is quite literally the overwhelming power just a few rocket based infantry have on them. Here are some numbers I found.

Here are the before 50 percent resistance and after 50 percent resistance (Number of rockets the "tanks" can soak from an ungarrisoned unit and before any upgrades):

ECM 8/15 (Could use some tweaking if needed)
Inferno 3/6
Battle Master: 10/20 (Nuke BM Untested). level 2 tank Battle Master from 13 to 26
Out Post 6/12
Vanilla Overlord: 28/55
Nuke Cannon 6/12
Vanilla Troop Crawler 6/12 (Attack troop crawler unaffected)
Paladin 13/24 (May need a price increase or PDL rate of fire decrease to not be overpowered vs GLA's)
Microwave Tank 12/24 (This is especially good).
Crusader 12/24
Scorpion 10/19
Marauder 13/25
Scud launcher (untested)

Here we see the only units that are affected by the proposed change. It is important to remember that we can specify any units listed that we do not want to have extra resistance.  Furthermore, we can look into other changes to compensate for the added resistance to rocket based infantry in needed.

Other Findings:

-When pairing up a 1 on 1 with a tank hunter and a battle master, tank hunter won convincingly with 1/3 of its health. 
-Rocket based infantry are still THE counter to tanks, however this change gives room for tanks to be micromanaged and fill the role of soaking when mixed with other units.
-Garrison units deal 20 percent more damage for some reason. This including when they are inside of bunker overlord/lixes, busses, and vees. 
-It seems laser lock from Missile Defenders are unaffected by the change.
-A full lix killed a tank in less than two full strikes. Now it is a little less than four strikes. (Due to the 20 percent damage bonus)

In theory, tanks would NOT slow down gameplay, but instead find a role to be unit mixed and used to break well-fortified enemy lines. (GLA tunnels mixed with civilian buildings).

Rocket based infantry would still completely obliterate everything else, including structures and non-tank based units, which would maintain the current meta's validity.

This needs LOTS more testing, and I would be happy to help. I have the INI folder changes already good to test, and hopefully will be learning how to create it as a map INI for everyone to easily test.

**Comment by xezon** [comment]
I won't be able to be at service for testing.

**Comment by fastAiRR** [comment]
> I won't be able to be at service for testing.

what does this mean

**Comment by xezon** [comment]
It means I am busy and have no time for testing.

**Comment by KrewGR** [comment]
Well, if anyone who reads this sees this, I literally have a map that has the changes. All we have to do is spend some time playing it to test anything we want.

**Comment by KrewGR** [comment]
So, I managed to get a few test games to test the most important match ups that this would affect. The findings help A LOT as now I have a direction as to what we can potentially do. 

two match ups were most important I think, and that is tank vs stealth (our balance standard) and inf vs tank (our extreme imbalance standard).

Inf vs tank was certainly better, and the matches were far more back and forth, but after enough games, it was clear inf was still the choice to have a higher win percentage, however, despite that, I think 50 percent is too much over all for the game, because inf vs tank is just a design flaw all together. Dragon/TH spam was still the preferred method as those were unchanged and still performed better than using Battle Masters.

Stealth vs tank was certainly much more difficult this time, and although I am not an expert, seeing 4 of my RPGS (one of them level 2) not even kill one level two tank in a group of 5 tanks was enough for me to be convinced that 50 percent resistance to infantry rockets was 
 far too much. 

What I will do now is test new changes as soon as I can based on these findings.

The changes I will test:

Tank Armor resistance to infantry rockets 30%
Nuke and Tank general Tank hunter price reduction to 300
slight decrease in production time for stealth hijacker. 

If these changes are a positive for stealth vs tank and inf vs tank, then I will test all other match ups.

I have a feeling I will land at tank armor resistance to 20 percent by the end of all this. Also, hopefully none of these changes will slow down gameplay, or "feel" too different.

**Comment by ExiLe-ZH** [comment]
While I love the idea of Tanks being better and having more utility in this game, a 100% armor buff will change the game MASSIVELY, it opens up a can of worms for many matchups. Just a small example, how is GLA now gonna deal with Paladin drops? It's already tough to deal with and with this change Crusaders can fill this role aswell.

One idea I much prefer when it comes to buffing Tanks is make them vet faster and give off less XP, it doesn't change the base balance too much, but it rewards good micro and it'll be less unforgiving to lose tanks.

**Comment by KrewGR** [comment]
You are correct about the 50 percent boost being too much. Will test more with smaller armor boosts and micro changes. I can test both ideas out as well, but the problem we will run into is that often, tanks don't kill anything before their death.

The paladin may see it's PDL rate of fire nerfed if proven too effective.

I think I will test the vet idea first. It seems safest.

**Comment by KrewGR** [comment]
Quick update. I did some basic testing with 20 percent armor resistance against rocket-based infantry. I think this is the sweet spot for now. All matchups will need testing, but it seems to work out great in theory for several reasons. 

1: When infantry is garrison, they receive a 20 percent increase in dealing damage for some reason. So, when Chinas are trying to fight GLAs (or any faction) with civilian building occupied, or when battlebus/lix/vees attack, Tanks will receive the same damage as if they were being shot by ungarrisoned units before the changes were applied. 

2. To maintain stealth vs tank balance and overall balance of antitank infantry, I changed infantry resistance to tank shells from 90 percent to 92 percent. Especially when considering the effect of upgrades such as speaker tower on overlords, veterancy on tanks and composite armor. 

3. In theory, this should help all Chinas (and tanks) with heavily fortified GLAs, but not break any match ups. 

4. Paladin is another concern, as there is a bug fix where composite armor works properly with the pala tank. I will be testing the idea of reducing PDL rate of fire on the paladin tank. 

5. I will also explore the idea of reducing vanilla battle master's cost to 650, as these units won't have a place despite the other changes. 

6. I will also explore the idea of increasing the cost of the emperor tank due to the numerous buffs it has received to keep balance vs GLA intact. 

If these tests go well, I will present it to everyone for advocation to make it an official part of the changes. If it doesn't, I will explore other alternatives to make tanks useful in ZH.

**Comment by KrewGR** [comment]
Another update: I tested paladin PDL rate of fire decrease to half. This showed good potential to not let USA paladin drops be OP vs GLAs, given the armor boost tanks will receive. However, overall the paladin performance was actually not good in initial testing, so I will test the price reduction to $1000. Furthermore, it will be likely that paladins will combo with avengers, since they are great support units for tanks.
-edit-
Also testing price reducing vanilla crusader to 850 and vanilla overlord to 1850

-edit 2- (update)

I will also test the microwave tank speed of disabling structures faster. I am aware that usa unit mixing could very well improve with the total amount of these ideas. Ideally, I expect players during testing phase to unit mix microwave tanks, crusader/paladins, avengers and tomahawks with some rocketvees, especially vs the GLAs. 

I'm thinking to leave the price of paladins at 1100 in the microwave buff, despite the PDL rate of fire decreased by 100 percent to maintain as much balance as possible vs the GLAs when playing as USA. I think paladins will be the preferred unit in these matchups, where as crusaders (hopefullly) will be the preferred tank vs some of the Chinas. (Tank, nuke, vchina).

Despite all this, the rocket vees are unchanged, and is very possible that players still won't bother with any of these test changes as rocket vees are still insanely powerful.

maybe it would be a decent idea to test price increases to certain USA vees, but who know right now. I don't want to touch anything that has nothing to do with helping the tanks find a place.

**Comment by KrewGR** [comment]
OK, here is the total and official list of changes I will be testing. After much thought, I think this will be a good place to start the 2nd testing attempt.

The changes:
-All tanks in game given 20 percent resistance to rocket based infantry. 
-Infantry resistance to tank shells from 90 percent to 92 percent (to help maintain the anti-tank infantry role, especially with stealth General)
-Stealth Hijacker production time decreased from 15 to 12. (Also to help with keeping balance between nuke/tank vs stealth)
-Paladin tank PDL rate of fire decreased by 100 percent (from 1 second to 2 seconds (To keep from Paladin drops from being overpowered. I also expect to see players to unit mix avengers with paladins)). 
-Vanilla Crusader tank price decreased to 800
-Vanilla battle master price reduced to 650
-Vanilla overlord price reduced to 1850
-Microwave tank disable speed to structures and defenses from 50 to 200 (It is not instant, but it is a lot faster. Range is still the same).
-China tank Emperor cost to 2200 (to also help maintain balance vs the GLAs)

Some notes to consider:

-This test is also taking account of the changes that have already been approved in the upcoming patch. This includes the subliminal messaging upgrade bug to emperor tanks being fix, ECM tank auto targeting bug, and composite armor bug for the Paladin. Although, this test version does not include any of those yet. 
-Just like before, garrison units still deal 20 percent more damage when inside civilian buildings, busses, lixes, China bunkers (and possibly vees).
-Laser Lock is unaffected as far as damage testing goes vs tanks. (Can be changed if needed).

**Comment by ExiLe-ZH** [comment]
I would also consider looking into some other parameters when it comes to tanks, like (turning) speed and engagement ranges.

**Comment by ReLaX82** [comment]
I like some ideas, but think it can be too drastic.
